backend/src/config/mod.rs

```diff
@@ -229,29 +416,23 @@ pub async fn set_dry(
     let mut db = ctx.db.handle();
     let mut tx = db.begin().await?;
     let mut breakages = BTreeMap::new();
-    configure(
-        &ctx,
-        &mut tx,
-        &id,
-        config,
-        &timeout,
-        true,
-        &mut BTreeMap::new(),
-        &mut breakages,
-    )
-    .await?;
-    crate::db::DatabaseModel::new()
-        .package_data()
-        .idx_model(&id)
-        .expect(&mut tx)
-        .await?
-        .installed()
-        .expect(&mut tx)
-        .await?
-        .status()
-        .configured()
-        .put(&mut tx, &true)
+    let locks = ConfigLocks::new(&mut tx).await?;
+    {
+        configure(
+            &ctx,
+            &mut tx,
+            &id,
+            config,
+            &timeout,
+            true,
+            &mut BTreeMap::new(),
+            &mut breakages,
+            &locks,
+        )
         .await?;
+    }
+
+    locks.configured.set(&mut tx, true, id.to_string()).await?;
```

Comment:
Still haven't figured out how to not convert to string for this. Originally I was playing with the 'static lifetime but that wouldn't work...